Buildings/Fluid/HeatExchangers/DXCoils/BaseClasses/CoilDefrostCapacity.mo

```diff
@@ -0,0 +1,257 @@
+
+  Modelica.Blocks.Interfaces.RealOutput PCra(
+    final unit="W",
+    displayUnit="W",
+    final quantity="Power")
+    "Crankcase heater power consumption"
+    annotation (Placement(transformation(extent={{100,70},{120,90}}),
+      iconTransformation(extent={{100,70},{120,90}})));
+
+  Real defMod_T
+    "Defrost modifier curve value";
+
+  Real RTF(
+    final unit="1",
+    displayUnit="1")
+    "Run-time fraction";
+
+  Real PLR(
+    final unit="1",
+    displayUnit="1")
+    "Part-load ratio";
+
+  Real PLFra(
+    final unit="1",
+    displayUnit="1")
+    "Part-load fraction";
+
+  Modelica.Units.SI.ThermodynamicTemperature TConInWetBul
+    "Wet bulb temperature of air entering indoor condenser coil";
+
+  Buildings.Utilities.Psychrometrics.TWetBul_TDryBulXi wetBul(
+    redeclare package Medium = MediumA)
+    annotation (Placement(transformation(extent={{-60,-30},{-40,-10}})));
+
+equation
+  TConInWetBul=wetBul.TWetBul
+    "Find wet-bulb temperature of inlet air";
+  defMod_T =Buildings.Utilities.Math.Functions.smoothMax(
+    x1=Buildings.Utilities.Math.Functions.biquadratic(
+      a=defCur.defEIRFunT,
+      x1=Modelica.Units.Conversions.to_degC(TConInWetBul),
+      x2=Modelica.Units.Conversions.to_degC(TOut)),
+      x2=0.001,
+      deltaX=0.0001)
+      "Cooling capacity modification factor as function of temperature";
+  PLR = uSpe;
+  if defOpe == Buildings.Fluid.HeatExchangers.DXCoils.BaseClasses.Types.DefrostOperation.resistive then
+    QDef = 0;
+    PDef =QDefResCap*tDefFra*RTF;
+  else
+    QDef =0.01*tDefFra*(7.222 - Modelica.Units.Conversions.to_degC(TOut))*QTot/1.01667;
+    PDef =defMod_T*QTot*tDefFra*RTF/1.01667;
+  end if
+  "Power and heat transfer calculations from EPlus engineering reference";
+  QTotDef = QTot * heaCapMul;
+  PLFra = Buildings.Fluid.Utilities.extendedPolynomial(
+      x=PLR,
+      c=defCur.PLFraFunPLR,
+      xMin=0,
+      xMax=1)
+    "Part load fraction for heating coil";
+  PTot = QTot * EIR * PLR * inpPowMul/PLFra;
+  RTF = PLR/PLFra;
+  PCra = defCur.QCraCap * (1 - RTF);
+
+  connect(pIn, wetBul.p)
+    annotation (Line(points={{-110,-50},{-70,-50},{-70,-28},{-61,-28}},
+      color={0,0,127}));
+  connect(XConIn, wetBul.Xi[1])
+    annotation (Line(points={{-110,-20},{-61,-20}}, color={0,0,127}));
+  connect(TConIn, wetBul.TDryBul)
+    annotation (Line(points={{-110,10},{-70,10},{-70,-12},{-61,-12}},
+                                                                   color={0,0,127}));
+   annotation (
+    defaultComponentName="defCap",
+    Icon(coordinateSystem(preserveAspectRatio=true, extent={{-100,-100},{100,100}}),
+                    graphics={
+        Text(
+          extent={{-100,100},{100,-100}},
+          textColor={0,0,255},
+          textStyle={TextStyle.Italic},
+          textString="f(To,Xo)")}),
+          Documentation(info="<html>
+Block to calculate heat transfered to airloop <code>QTotDef</code>, as well as 
```

Comment:
Needs to start with `<p>`, and there is no airloop in Modelica. That is an EnergyPlus-thing.